src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/operation/ExecuteTaskReconfigurationService.java

```diff
@@ -124,8 +124,8 @@ private List<String> reconfigureTasks(List<String> taskIds, List<TaskResource> s
                 try {
                     log.info("Re-configure task-id {}", taskId);
                     Optional<TaskResource> optionalTaskResource = cftTaskDatabaseService
-                        .findByIdAndObtainPessimisticWriteLock(taskId);
-
+                         .findByIdAndStateInObtainPessimisticWriteLock(taskId, List.of(CFTTaskState.ASSIGNED,
+                                                                                      CFTTaskState.UNASSIGNED));
                     if (optionalTaskResource.isPresent()) {
```

Comment:
In the case where we initially want to Reconfigure a taskId but then don't want to execute because the state has changed we should output logs giving at least a description which includes the taskId.  It may even be beneficial to include the caseId and state of the Task.
E.g. For <OperationName> did not execute for Task Resource: taskId: {}, caseId: {}, state {}

This will allow us to track in case we need to.

However this may mean that you have to either go and get more info from the Task record with another DB query or your original DB query returns the full Task.

We've seen in production that this is a rare use case so querying the DB again would be acceptable.